cli/src/migrate.js

```diff
@@ -0,0 +1,354 @@
+  const project = this.options.project_name;
+  return Promise.resolve().then(() => {
+    white('Removing suffix from user tables');
+    return r.db(`${project}_internal`).table('collections')
+      .forEach((collDoc) => r.db('rethinkdb').table('table_config')
+          .filter({ db: project, name: collDoc('table') })
+          .update({ name: collDoc('id') }))
+      .run(this.conn)
+      .then(() => green(' └── Suffixes removed'));
+  });
+}
+
+function moveInternalTables() {
+  // find project_internal
+  // move all tables from ${project}_internal.${table} to ${project}.hz_${table}
+  //   - except for users, don't add hz_prefix, but move its db
+  const project = this.options.project_name;
+  return Promise.resolve().then(() => {
+    white(`Moving internal tables from ${project}_internal to ${project}`);
+    return r.db('rethinkdb').table('table_config')
+      .filter({ db: `${project}_internal` })
+      .update((table) => ({
+        db: project,
+        name: r.branch(
+          table('name').ne('users'),
+          r('hz_').add(table('name')),
+          'users'),
+      })).run(this.conn)
+      .then(() => green(' ├── Internal tables moved'));
+  }).then(() => {
+    // delete project_internal
+    green(` └── Deleting empty "${project}_internal" database`);
+    return r.dbDrop(`${project}_internal`).run(this.conn);
+  });
+}
+
+function renameIndices() {
+  // for each user $table in ${project}
+  //    for each index in ${table}
+  //        parse the old name into array of field names.
+  //        rename to `hz_${JSON.stringify(fields)}`
+  const project = this.options.project_name;
+  return Promise.resolve().then(() => {
+    white('Renaming indices to new JSON format');
+    return r.db(project).tableList().forEach((tableName) => {
+      return r.db(project).table(tableName).indexList().forEach((indexName) => {
+        return r.db(project).table(tableName).indexRename(indexName, rename(indexName));
+      });
+    }).run(this.conn)
+    .then(() => green(' └── Indices renamed.'));
+  });
+
+  function rename(name) {
+    // ReQL to rename the index name to the new format
+    const initial = {
+      escaped: false,
+      field: '',
+      fields: [ ],
+    };
+    return name.split('')
+      .fold(initial, (acc, c) => {
+        return r.branch(
+          acc('escaped'),
+            acc.merge({
+              escaped: false,
+              field: acc('field').add(c),
+            }),
+          c.eq('\\'),
+            acc.merge({ escaped: true }),
+          c.eq('_'),
+            acc.merge({
+              fields: acc('fields').append(acc('field')),
+              field: '',
+            }),
+          acc.merge({ field: acc('field').add(c) })
+        );
+      })
+      // last field needs to be appended
+      .do((x) => x('fields').append(x('field')))
+      .toJSON()
+      .do((x) => r('hz_').add(x));
+  }
+}
+
+function rewriteHzCollectionDocs() {
+  // for each document in ${project}.hz_tables
```

Comment:
Whoops, these line notes were from the original sketch, I didn't update them after I wrote the code